src/main/java/org/broadinstitute/hellbender/tools/walkers/sv/JointGermlineCNVSegmentation.java

```diff
@@ -0,0 +1,483 @@
+                if (sampleDB.getSample(sample) == null) {
+                    logger.warn("Sample " + sample + " is missing from the supplied pedigree file. Will attempt to infer ploidy from segments VCF genotype ploidy."); //note that this only works if there are called variants on the sex chromosomes
+                }
+            }
+        }
+
+
+        dictionary = getBestAvailableSequenceDictionary();
+        //dictionary will not be null because this tool requiresReference()
+
+        final GenomeLocParser parser = new GenomeLocParser(this.dictionary);
+
+        if (modelCallIntervalList == null) {
+            callIntervals = null;
+        } else {
+        final List<GenomeLoc> inputCoverageIntervals = IntervalUtils.featureFileToIntervals(parser, modelCallIntervalList.getAbsolutePath());
+        final List<GenomeLoc> inputTraversalIntervals = IntervalUtils.genomeLocsFromLocatables(parser,getTraversalIntervals());
+            callIntervals = IntervalUtils.mergeListsBySetOperator(inputCoverageIntervals, inputTraversalIntervals, IntervalSetRule.INTERSECTION);
+        }
+
+        defragmenter = new SVDepthOnlyCallDefragmenter(dictionary, 0.8, callIntervals);
+        clusterEngine = new SVClusterEngine(dictionary, true, breakpointSummaryStrategy);
+
+        vcfWriter = getVCFWriter();
+    }
+
+    //TODO: this is the third copy of this method
+    /**
+     * Entry-point function to initialize the samples database from input data
+     */
+    private SampleDB initializeSampleDB() {
+        final SampleDBBuilder sampleDBBuilder = new SampleDBBuilder(PedigreeValidationType.STRICT);  //strict will warn about missing samples
+        if (pedigreeFile != null) {
+            sampleDBBuilder.addSamplesFromPedigreeFiles(Collections.singletonList(pedigreeFile));
+        }
+        return sampleDBBuilder.getFinalSampleDB();
+    }
+
+    private VariantContextWriter getVCFWriter() {
+        samples = getSamplesForVariants();
+
+        final VCFHeader inputVCFHeader = new VCFHeader(getHeaderForVariants().getMetaDataInInputOrder(), samples);
+
+        final Set<VCFHeaderLine> headerLines = new LinkedHashSet<>(inputVCFHeader.getMetaDataInInputOrder());
+        headerLines.addAll(getDefaultToolVCFHeaderLines());
+        headerLines.add(GATKSVVCFHeaderLines.getInfoLine(GATKSVVCFConstants.SVLEN));
+        headerLines.add(GATKSVVCFHeaderLines.getInfoLine(GATKSVVCFConstants.SVTYPE));
+        headerLines.add(VCFStandardHeaderLines.getInfoLine(VCFConstants.ALLELE_FREQUENCY_KEY));
+        headerLines.add(VCFStandardHeaderLines.getInfoLine(VCFConstants.ALLELE_COUNT_KEY));
+        headerLines.add(VCFStandardHeaderLines.getInfoLine(VCFConstants.ALLELE_NUMBER_KEY));
+
+        VariantContextWriter writer = createVCFWriter(outputFile);
+
+        final Set<String> sampleNameSet = new IndexedSampleList(samples).asSetOfSamples();
+        final VCFHeader vcfHeader = new VCFHeader(headerLines, new TreeSet<>(sampleNameSet));
```

Comment:
I copied this from a tool that copied this from a tool that copied this from CombineGVCFs where it has @jamesemery 's name on it.  I looks like these hijinx are to prevent the writer's samples from being modified?